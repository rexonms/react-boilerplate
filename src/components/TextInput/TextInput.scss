.container {
  .label {
    font-family: $fontFamilyDefault;
    font-weight: normal;
    font-size: $fontSizeDefault;
    line-height: $lineHeightDefault;
    padding-bottom: 5px;
    display: block;
    text-transform: uppercase;
    position: relative;
    cursor: pointer;
    color: gray;
  }
  input {
    display: block;
    outline: 0;
    border: 1px solid #ccc;
    transition: background-color 200ms linear;
    padding: 10px;
    width: 100%;
    margin: 0;
    font-size: 16px;
    background-color: #f7f7f7;
  }
  .inputError {
    border-color: #c00;
  }
  .error {
    font-family: $fontDefault;
    text-align: right;
    font-size: x-small;
    color: #c00;
  }
}