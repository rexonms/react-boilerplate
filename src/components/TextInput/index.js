import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './TextInput.scss';

class Input extends Component {
  constructor(props) {
    super(props)
    this.state = {
      value: this.props.value,
    };

    this.onChangeHandler = this.onChangeHandler.bind(this);
  }

  onChangeHandler(e) {
    this.setState({
      value: e.target.value,
    });
  }
  render() {
    return (
      <div id={this.props.id} className={styles.container}>
        <label htmlFor={this.props.id} className={styles.label}>
          {this.props.labelText}
          <input
            className={styles.input}
            name={this.props.id}
            value={this.state.value}
            placeholder={this.props.placeholder}
            onChange={this.onChangeHandler}
          />
        </label>
      </div>
    );
  }
}

Input.defaultProps = {
  value: '',
  placeholder: '',
  onChangeHandler: () => {}
}

Input.propTypes = {
  /** Unique id for the input container */
  id: PropTypes.string.isRequired,
  /** Label for the input box */
  labelText: PropTypes.string.isRequired,
  /** Pre filled value for the input box, if applicable */
  value: PropTypes.string,
  placeholder: PropTypes.string,
  /** onChangeHandler that updates the input box state */
  onChangeHandler: PropTypes.func,
}

export default Input;
