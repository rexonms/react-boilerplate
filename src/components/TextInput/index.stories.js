import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import TextInput from './index';

const stories = storiesOf('Components/TextInput', module);
stories.addDecorator(withKnobs);

// Knobs as dynamic variables.
stories.addDecorator(story => (story()));

stories.add('TextInput', withInfo(`
  Text fields enable the user to interact with and input data. A single line field is used when the input anticipated by the user is a single line of text as opposed to a paragraph.
  ~~~js
    <TextInput id="input-story" labelText="standard input"/>
  ~~~
`)(() => {
  return (
    <TextInput id="input-story" labelText="standard input"/>
  )}));

