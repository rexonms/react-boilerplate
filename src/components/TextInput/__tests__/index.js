import React from 'react';
import renderer from 'react-test-renderer';

import TextInput from './../index';

it('renders correctly', () => {
  const tree = renderer.create(
    <TextInput
      labelText="test table"
      id="unique id"
      onChangeHandler={() => {}}
    />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
