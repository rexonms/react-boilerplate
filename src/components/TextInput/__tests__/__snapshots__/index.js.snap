// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
  id="unique id"
>
  <label
    className="label"
    htmlFor="unique id"
  >
    test table
    <input
      className="input"
      name="unique id"
      onChange={[Function]}
      placeholder=""
      value=""
    />
  </label>
</div>
`;
